src/Logic.hs

```diff
@@ -272,16 +272,19 @@ handleCommentAdded
   -> Text
   -> ProjectState
   -> Action ProjectState
-handleCommentAdded triggerConfig projectConfig pr author body =
+handleCommentAdded triggerConfig projectConfig pr author body state =
   let
     -- If the message was a valid merge command, then it was approved by the
     -- author of the message. Otherwise do nothing.
-    update pullRequest =
-      if (isMergeCommand triggerConfig body) && (isReviewer projectConfig author)
-        then pullRequest { Pr.approvedBy = Just author }
-        else pullRequest
+    update pullRequest = pullRequest { Pr.approvedBy = Just author }
+    isValidApproveCommand = (isMergeCommand triggerConfig body) && (isReviewer projectConfig author)
   in
-    pure . Pr.updatePullRequest pr update
+    if isValidApproveCommand
```

Comment:
This will conflict with #20, because `isReviewer` is needs to run in the monad. But it should be easy to fix.